File: 1/1.rs
/* Advent of Code */
use std::fs;

fn main() {
    let data = fs::read_to_string("input").expect("Failed to read file");
    let mod_fuel: Vec<f64> = data.lines() // split by '\n'
                                 .map(|x| // convert masses to their fuels
                                     extra_fuel(x.parse::<f64>().unwrap())
                                 )
                                 .collect();

    let total_fuel: f64 = mod_fuel.iter().sum();
    println!("{:?}", total_fuel);
}

fn calculate_fuel(module: f64) -> f64 {
    (module / 3.0).trunc() - 2.0
}

fn extra_fuel(fuel: f64) -> f64 {
    if calculate_fuel(fuel) <= 0.0 {
        0.0
    } else {
        calculate_fuel(fuel) + extra_fuel(calculate_fuel(fuel))
    }
}

File: 1/1.py
def calculate_fuel(module):
    return (module // 3) - 2


def total_fuel(fuel):
    if calculate_fuel(fuel) <= 0:
        return 0
    else:
        return calculate_fuel(fuel) + total_fuel(calculate_fuel(fuel))


def main():
    try:
        with open("input.txt", "r") as f:
            print(sum(
                [total_fuel(int(line.strip())) for line in f.readlines()]))
    except FileNotFoundError:
        print("'input.txt' was not found")


if __name__ == '__main__':
    main()
